provider/pkg/provider/provider.go

```diff
@@ -210,8 +210,135 @@ func (k *kubeProvider) Call(ctx context.Context, req *pulumirpc.CallRequest) (*p
+		"kubernetes_pod_security_policy":                 "kubernetes:policy/v1beta1:PodSecurityPolicy",
+		"kubernetes_pod_security_policy_v1beta1":         "kubernetes:policy/v1beta1:PodSecurityPolicy",
+		"kubernetes_priority_class":                      "kubernetes:scheduling.k8s.io/v1alpha1:PriorityClass",
+		"kubernetes_priority_class_v1":                   "kubernetes:scheduling.k8s.io/v1:PriorityClass",
+		"kubernetes_replication_controller":              "kubernetes:core/v1:ReplicationController",
+		"kubernetes_replication_controller_v1":           "kubernetes:core/v1:ReplicationController",
+		"kubernetes_resource_quota":                      "kubernetes:core/v1:ResourceQuota",
+		"kubernetes_resource_quota_v1":                   "kubernetes:core/v1:ResourceQuota",
+		"kubernetes_role":                                "kubernetes:rbac.authorization.k8s.io/v1alpha1:Role",
+		"kubernetes_role_v1":                             "kubernetes:rbac.authorization.k8s.io/v1:Role",
+		"kubernetes_role_binding":                        "kubernetes:rbac.authorization.k8s.io/v1alpha1:RoleBinding",
+		"kubernetes_role_binding_v1":                     "kubernetes:rbac.authorization.k8s.io/v1beta1:RoleBinding",
+		"kubernetes_runtime_class_v1":                    "kubernetes:node.k8s.io/v1:RuntimeClass",
+		"kubernetes_secret":                              "kubernetes:core/v1:Secret",
+		"kubernetes_secret_v1":                           "kubernetes:core/v1:Secret",
+		"kubernetes_service":                             "kubernetes:core/v1:Service",
+		"kubernetes_service_v1":                          "kubernetes:core/v1:Service",
+		"kubernetes_service_account":                     "kubernetes:core/v1:ServiceAccount",
+		"kubernetes_service_account_v1":                  "kubernetes:core/v1:ServiceAccount",
+		"kubernetes_stateful_set":                        "kubernetes:apps/v1beta2:StatefulSet",
+		"kubernetes_stateful_set_v1":                     "kubernetes:apps/v1:StatefulSet",
+		"kubernetes_storage_class":                       "kubernetes:storage.k8s.io/v1beta1:StorageClass",
+		"kubernetes_storage_class_v1":                    "kubernetes:storage.k8s.io/v1:StorageClass",
+		"kubernetes_token_request_v1":                    "kubernetes:authentication.k8s.io/v1:TokenRequest",
+		"kubernetes_validating_webhook_configuration":    "kubernetes:admissionregistration.k8s.io/v1beta1:MutatingWebhookConfiguration",
+		"kubernetes_validating_webhook_configuration_v1": "kubernetes:admissionregistration.k8s.io/v1:MutatingWebhookConfiguration",
+	}
+
+	resources := make(map[string]interface{})
+	for tftok, putok := range resourceTokens {
+		// Skip if the token is empty.
+		if putok == "" {
+			continue
+		}
+		resources[tftok] = map[string]interface{}{
+			"tok": putok,
+			"fields": map[string]interface{}{
+				"metadata": map[string]interface{}{
+					"maxItemsOne": true,
+				},
+				"spec": map[string]interface{}{
```

Comment:
`spec` isn't present on all k8s resources (`apiVersion`, `kind`, and `metadata` are the only guaranteed fields). I don't completely understand how this mapping is supposed to work, but I suspect this isn't right.

Reply:
👍 I'm reworking this. I thought this simple map would suffice but turns out it's trickier (isn't it always 🙄)